src/main/java/org/agrona/concurrent/ringbuffer/OneToOneRingBuffer.java

```diff
@@ -132,13 +133,15 @@ public boolean write(final int msgTypeId, final DirectBuffer srcBuffer, final in
 
         if (0 != padding)
         {
+            buffer.putLong(0, 0L);
```

Comment:
This is subtle and as you say later, initially seems redundant.

This line is required to maintain the invariant that the reader observes a subsequent zero record header after each record with length greater than zero, otherwise it's a race.

When the reader observes the non-zero padding record length written at line 137 below, then this zeroing of index 0 at line 136 guarantees that the reader also observes the subsequent zero record header.

Later, when the reader observes the non-zero message record length written at line 143 below, then the zeroing after the message record at line 142 guarantees that the reader also observes the subsequent zero record header.

If line 136 is omitted, then the reader may observe the non-zero padding length without a subsequent zero record header before the message record and the subsequent zero record header are observable. This would break the invariant for the padding record.